src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -1727,10 +1727,14 @@ private void WriteRedFactory(Node nd)
                 {
                     Write("{0}.Node.ToGreenList<Syntax.InternalSyntax.CSharpSyntaxNode>()", CamelCase(field.Name));
                 }
-                else
+                else if (IsOptional(field))
                 {
                     Write("{0} == null ? null : (Syntax.InternalSyntax.{1}){0}.Green", CamelCase(field.Name), field.Type);
```

Comment:
I don't know if you're doing this in a later commit, but these things can just be `?.`

Reply:
this case can't be unfortunately.  there's no `?.` syntax that is shorthand for `a == null ? null : (B)a.b`.  `(B)a?.B` is not equivalent as that forces the cast to always happen.